### ISSUE TITLE
When using Try It Out with Open API 3, multiple cookie apiKeys are separated with & not ; as required by curl

### ISSUE THREAD

**Issue by absurdist** [swagger-api/swagger-ui#4218]
When multiple API keys are specified in the security object, Try it Out separates the name value pairs with an ampersand. Curl expects a semi colon and the request will fail

I'm attempt to locate the source of the error, any guidance would be appreciated.

I'll leave the template below and fill it out tomorrow when I am back at work.

| Q                               | A
| ------------------------------- | -------
| Bug or feature request?         |  
| Which Swagger/OpenAPI version?  |
| Which Swagger-UI version?       |
| How did you install Swagger-UI? | 
| Which browser & version?        |
| Which operating system?         | 

### Demonstration API definition

```yaml
your: "API definition goes here"
```

### Configuration (browser query string, constructor, config.yaml)

```js
{
  "your": { "constructorConfig": "here" }
}
```

`?yourQueryStringConfig=here`

**Comment by sologub-s** [comment]
I have the same issue.

**Comment by Raptor399** [comment]
There is a second issue at play here: the apiKey values are being UriEncoded, but they shouldn't be. Granted, it's vague [what characters cookies allow](https://stackoverflow.com/questions/1969232/allowed-characters-in-cookies), but I see cookie values unescaped in in the request the latest version of Chrome, so I'm guessing that is the standard.

For example:

```
curl -X GET "http://localhost/tests" -H "accept: */*" -H "Cookie: key-a=a%2Ba&keyB=b%2Fb"
```

should be:

```
curl -X GET "http://localhost/tests" -H "accept: */*" -H "Cookie: key-a=a+a; keyB=b/b"
```

Please take this into account when fixing this bug.

**Comment by Raptor399** [comment]
Update: judging by the reaction to #OAI/OpenAPI-Specification#1676, apiKey values in the cookie are wrongly being UriEncoded by swagger-ui. From the issue:

> Currently we don't limit the characters allowed in security parameters

This means there is no reason to UriEncode apiKey parameter values in the cookie according to the spec, so please don't encode them in swagger-ui (as is currently the case).

**Comment by shockey** [comment]
Catching up here - are we all talking about the cURL output, or is this being observed in the requests Swagger UI itself is sending out as well?

**Comment by Raptor399** [comment]
@shockey I observed UriEncoded apiKey cookie values in Swagger UI in Chrome at first. Calls would not work in the browser, so I copied the curl command to a terminal for further testing. There I discovered that undoing UriEncoding of the apiKey cookie values made things work as expected.

**Comment by amitbaraik** [comment]
Hello All,

do anyone know how to fix this? My problem is same as absurdist

**Comment by t1bb4r** [comment]
Anyone have a workaround? Issue has been open for over a year. How are you using cookies?

**Comment by t1bb4r** [comment]
Maybe to clarify the title, it is not required by curl it is required by http (all http web servers), https://tools.ietf.org/html/rfc6265#section-5.4
https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cookie
"A list of name-value pairs in the form of `<cookie-name>=<cookie-value>`. Pairs in the list are separated by a semicolon and a space (`'; '`)."

**Comment by t1bb4r** [comment]
Sorry I'm stupid, my cookies were not being set due to an unrelated issue. I thought this was the reason, but I see now that this is indeed just an issue with how curl is formatted and not an issue with the functionality of cookies.

**Comment by aantillonl** [comment]
Also having this issue. As a workaround, and pretty ugly one, you could instead defining a Security Scheme with `in: cookie` for the API Key, define it as a Parameter for your operation as follows:
```
parameters:
        - name: authCookie
          schema:
            type: object
            properties:
              cookie1:
                type: string
              cookie2:
                type: string                  
          in: cookie
          explode: true
          style: matrix
```
This would generate a curl command with the header `Cookie= ;cookie1=value;cookie2=value` which at least for me worked from the terminal. Note that the cookie values are not formatted when you use this method(i.e. symbols like "+" and "/', are inserted unencoded)

The disadvantage is that you need to define this params **for every** route. Also the cookie values are defined as json in a text area from the UI instead of the nicer form available for Security Schemes.

**Comment by till213** [comment]
To add that I just stumbled over this same problem: we use "XML formatted cookies" (SAML assertions). For the given request the developer tool "Postman" generates the following command:

```
curl --location --request POST 'http://example.com:/my/service/' \
--header 'Content-Type: application/json' \
--header 'Cookie: SAML_ASSERTION="<saml2:Assertion xmlns:saml2=' ... </saml2:Assertion>"' \ ...
```

However when selecting "Try it out" in SwaggerUI (within Visual Studio Code, in case that matters) the generated curl command is like:

`curl -X POST "http://example.com:/my/service/" -H  "accept: application/json" -H  "Content-Type: application/json" -H  "Cookie: SAML_ASSERTION=%3Csaml2%3AAssertion%20xmlns%3Asaml2%3D'urn ... sertion%3E" -d ...`

In other words: the "XML cookie" gets "URI encoded".

Would be nice if this issue would be fixed ;) +1 from me anyway...

**Comment by bam-bz** [comment]
This looks like the location where the ampersand separator is added:
https://github.com/swagger-api/swagger-js/blob/84bff85dec3a4a7e70db3fc7fbe67154c51418bf/src/execute/index.js#L292

I'm not familiar enough with swagger to know whether replacing `&` with `; ` on that line would break any valid use cases.
